src/app/credential-internal.ts

```diff
@@ -421,6 +432,17 @@ function requestAccessToken(client: HttpClient, request: HttpRequestConfig): Pro
   });
 }
 
+/**
+ * Obtain a new OIDC token by making a remote service call.
+ */
+function requestIDToken(client: HttpClient, request: HttpRequestConfig): Promise<string> {
+  return client.send(request).then((resp) => {
+    return resp.text || '';
```

Comment:
Good call - we probably should throw in this case (since if  were requesting an id token & get an empty string, we'll almost certainly get an error later).